### ISSUE TITLE
A new solution for issue #67: Uncaught ReferenceError: exports is not defined

### ISSUE THREAD

**Issue by codingisnothingtodo** [PhaserEditor2D/PhaserEditor#70]
I think I've got an exact solution for modularity problem of Typescript projects. I wrote a Java class that includes imported, extended and dependency classes into Level.js file. But user must define its class hierarchy in an XML file. So I think you may want to look and discuss about this. If you like my code you may want to develop more accurate version and integrate it to your new product. Because according to my research I understand that Typescript has some modularity problems because of compiling into Javascript and Javascript whose version is older than Ecma 2015 doesn't support modules natively  Except it may work correctly in its native IDE, Visual Studio that I haven't tried it.
[moduleTrial.zip](https://github.com/PhaserEditor2D/PhaserEditor/files/1684976/moduleTrial.zip)
[shapeControllerBuild.zip](https://github.com/PhaserEditor2D/PhaserEditor/files/1684977/shapeControllerBuild.zip)

**Comment by PhaserEditor2D** [comment]
Hi, right now I am very busy with the coming v1.5.0 release, after this release, I should do a huge re-write of the editor to support phaser v3 and other new stuff I have in mind. I will look to this issue then. Probably this could be added via a third-party plugin.

**Comment by PhaserEditor2D** [comment]
By the way, did you try by compiling all TS files into a single JS file? The TypeScript compiler has that option.

**Comment by codingisnothingtodo** [comment]
Yes Arian I tried it. But there was so many errors. And also I used the
option at tsconfig.json.

2018-02-13 1:06 GMT+03:00 Phaser Editor <notifications@github.com>:

**Comment by bfountaine** [comment]
I've got a much simpler solution to this. If I understand you correctly what you want to do is:

export class MyClass {
 // class properties and methods here.
}

so that in another class file you can go

import {MyClass } from "./MyClass";

export class MyOtherClass extends MyClass {

}

what you can do instead is put the export statement on the bottom line eg:

class MyClass {
 // class properties and methods here.
}
export { MyClass }

then your 2nd class can be the same as above.
